portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,106 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  Comment comment = new Comment();
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = getParameter(request, "text-input", "");
+    long timestamp = System.currentTimeMillis();
+    boolean upperCase = Boolean.parseBoolean(getParameter(request, "upper-case", "false"));
+    boolean sort = Boolean.parseBoolean(getParameter(request, "sort", "false"));
+
+    // Convert the text to upper case.
+    if (upperCase) {
+      text = text.toUpperCase();
+    }
+
+    // Set the Comment object
+    comment.setComment(text);
+
+    // Make datastore and entity 
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity inputEntity = new Entity("Input");
+    inputEntity.setProperty("timestamp", timestamp);
+    if (comment.getComment() != null) {
+        // if comment is not null
+        inputEntity.setProperty("text", comment.getComment());
+    }
+     // Store the entity
+    datastore.put(inputEntity);
+
+    // Redirect to /index.html 
+    response.setContentType("text/html;");
+    response.sendRedirect("/index.html");
+   
+  }
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Alex!</h1>");
+
+    // Get comments from datastore 
+    Query query = new Query("Input").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<String> comments = new ArrayList<>();
+
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String text = (String) entity.getProperty("text");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      comments.add(text);
+    }
+    // Print comments out
+    for (String comm : comments) {
+        response.getWriter().println("> " + comm); 
+
+    }
+    // response.getWriter().println(comments); 
+  }
+
+  /**
+   * Converts an ArrayList<String> instance into a JSON string using the Gson library. Note: We first added
+   * the Gson library dependency to pom.xml.
+   */
+  private String convertToJsonUsingGson(HttpServletResponse response) {
+    Gson gson = new Gson();
+    String json = gson.toJson(response);
+    return json;
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
```

Comment:
This can also be written as 

`return request.getParameter(name) == null ? defaultValue : value;`